rotkehlchen/chain/ethereum/modules/balancer/utils.py

```diff
@@ -0,0 +1,284 @@
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    token0 = get_ethereum_token(
+        symbol=token0_symbol,
+        ethereum_address=attr_addresses['tokenIn'],
+        name=token0_name,
+        decimals=token0_decimals,
+    )
+    token1 = get_ethereum_token(
+        symbol=token1_symbol,
+        ethereum_address=attr_addresses['tokenOut'],
+        name=token1_name,
+        decimals=token1_decimals,
+    )
+    amm_swap = AMMSwap(
+        tx_hash=tx_hash,
+        log_index=log_index,
+        address=attr_addresses['userAddress'],
+        from_address=attr_addresses['caller'],
+        to_address=attr_addresses['poolAddress'],
+        timestamp=timestamp,
+        location=Location.BALANCER,
+        token0=token0,
+        token1=token1,
+        amount0_in=amount0_in,
+        amount1_in=AssetAmount(ZERO),
+        amount0_out=AssetAmount(ZERO),
+        amount1_out=amount1_out,
+    )
+    return amm_swap
+
+
+def deserialize_token_price(
+        raw_token_price: Dict[str, Any],
+) -> Tuple[ChecksumEthAddress, Price]:
+    """May raise DeserializationError"""
+    try:
+        token_address = raw_token_price['id']
+        usd_price = deserialize_price(raw_token_price['price'])
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    try:
+        token_address = to_checksum_address(token_address)
+    except ValueError as e:
+        raise DeserializationError(
+            f'Invalid ethereum address: {token_address} in token price.',
+        ) from e
+
+    return token_address, usd_price
+
+
+def deserialize_token_day_data(
+        raw_token_day_data: Dict[str, Any],
+) -> Tuple[ChecksumEthAddress, Price]:
+    """May raise DeserializationError"""
+    try:
+        token_address = raw_token_day_data['token']['id']
+        usd_price = deserialize_price(raw_token_day_data['priceUSD'])
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    try:
+        token_address = to_checksum_address(token_address)
+    except ValueError as e:
+        raise DeserializationError(
+            f'Invalid ethereum address: {token_address} in token day data.',
+        ) from e
+
+    return token_address, usd_price
+
+
+def calculate_amm_trade_from_amm_swaps(swaps: List[AMMSwap]) -> AMMTrade:
+    """AMMTrade is always a buy.
+
+    [USDC -> AMPL]               BASE_QUOTE pair is AMPL_USDC.
+    [USDC -> AMPL, AMPL -> WETH] BASE_QUOTE pair is WETH_USDC.
+    """
+    amm_trade = AMMTrade(
+        trade_type=TradeType.BUY,
+        base_asset=swaps[-1].token1,
+        quote_asset=swaps[0].token0,
+        amount=swaps[-1].amount1_out,
+        rate=Price(swaps[-1].amount1_out / swaps[0].amount0_in),
+        swaps=swaps,
+        trade_index=0,
+    )
+    return amm_trade
+
+
+def get_trades_from_tx_swaps(
+        swaps: List[AMMSwap],
+        is_close_amount_mode: bool = False,
```

Comment:
[This transaction](https://etherscan.io/tx/0x70b50d011f23f8437aa32b4155972098b34b5fb19b02198f3bc7f2456c4c3ffc) is perfect for testing this mode. In a single transaction we have 3 swaps involving 3 different pools. With this mode enabled we get a single AMMTrade, otherwise 3 different trades.

Along this function tests there are 2 for that particular transaction, one with the mode enabled and one disabled.